### ISSUE TITLE
SSL for downloads site

### ISSUE THREAD

**Issue by coreyfarrell** [akheron/jansson#407]
I'm working on enabling SSL for everything in Asterisk.  One of the scripts downloads jansson from http://www.digip.org/jansson/releases/ to support building it for unpackaged build of jansson.  Would it be possible to enable SSL for this site?  I'm unsure of your server configuration but free certificates can be acquired from https://letsencrypt.org/.

I know you're busy but I hope you'll consider making this change.  The main problem is that I need a place to download release tarballs which includes the generated 'configure' and verify it.  Currently the downloads and signatures are only available from HTTP.

**Comment by isaachier** [comment]
Why don't you just use the release here on GitHub: https://github.com/akheron/jansson/releases?

**Comment by ploxiln** [comment]
The point of signatures is, if you have imported the correct Petri Lehtinen public key into your system, no one else can create a signature which that public key validates. So if you check the signatures correctly, then you don't need "secure" downloads of the tarballs nor the signatures, because a maliciously substituted signature would not be made with the correct public key.

**Comment by isaachier** [comment]
But how can you trust the signature posted on an HTTP site where the package and signature could be generated by a man in the middle attacker?

**Comment by ploxiln** [comment]
You're thinking of a hash, like sha256. An attacker can generate a hash of the malicious tarball, and substitute it the same was as the tarball. A "signature by Petri Lehtinen's private key" cannot be made by an attacker. And with Petri Lehtinen's public key, that's what you validate.

**Comment by coreyfarrell** [comment]
I'll have to look into the possibility of using GPG verification, I was thinking of SHA256 or similar hashes.  One issue is that this download is being done from a script and I'm not sure how to check it without manipulating the system itself.  I'll have to see if it's possible to verify a signature matches a public key that is not imported (check against Petri's public key stored in a file only).  I suppose I could use the GPG key to manually verify the file I download then generate an SHA hash to be included in our own tarball.

As for using the github releases they are not the same as the digip.org downloads, they are missing some files such as ./configure.  Adding autoconf as a dependency would be difficult.

**Comment by akheron** [comment]
Enabling HTTPS has actually been on my todo list for a while. The hosting of digip.org is not entirely in my own hands, but I'll see what I can do.

**Comment by GitMensch** [comment]
https://digip.org/jansson/releases/ is correctly setup with Let's encrypt - so this is another issue that can be closed.

And also for the very last release the tarballs and keys were uploaded in the release.
@akheron Maybe you can edit the old releases (from gdigip.org) and post-add at least the tarballs? To have the GitHub Releases as a mirror is a nice thing; it also is good to not _only_ have this, I'm currently working from an machine over SSH which seems to be blocked by GitHub (or its cdn is broken for that area, also possible).
